### ISSUE TITLE
Using <Upload> as file chooser for a Form field

### ISSUE THREAD

**Issue by maxkrieger** [ant-design/ant-design#6983]
### Version
2.12.2

### Environment
MacOS Sierra, Chrome 59

### Reproduction link
[https://codepen.io/a9/pen/NvGGyK?editors=0010](https://codepen.io/a9/pen/NvGGyK?editors=0010)

### Steps to reproduce
Prevent `action` uploading by returning `false` on `beforeUpload`. This results in blocking fieldDecorator from registering the file list.

### What is expected?
There should be a way to submit uploaded files as within the `FormData` of the fielddecorator when using `<Upload>`, rather than the file being immediately uploaded to a specific path.

### What is actually happening?
An error is thrown if you don't upload to a path and return false for `beforeUpload`, preventing this from being possible.

---
In summary, if your server would only accept the file as a field within a form/custom request (for security and convenience), the `Upload` component can't work as a "dumb" file chooser. It would be nice to have a way to use it as such instead of a direct uploader.

**Comment by benjycui** [comment]
@maxkrieger your CodePen doesn't work, please fix it first, thanks.

**Comment by afc163** [comment]
https://ant.design/components/upload/#components-upload-demo-upload-manually

**Comment by maxkrieger** [comment]
@afc163 How do you use with `formDecorator`?

**Comment by benjycui** [comment]
> @maxkrieger your CodePen doesn't work, please fix it first, thanks.

**Comment by maxkrieger** [comment]
@benjycui https://codepen.io/a9/pen/VzaaNj?editors=0011 see how value is not injected into `getFormDecorator` if `beforeUpload` returns false.

**Comment by benjycui** [comment]
@afc163 when `beforeUpload` return `false`, [upload logic](https://github.com/react-component/upload/blob/54255032ab4607070577db2e36647ee2a773b64f/src/AjaxUploader.jsx#L86-L108) will not be executed, so `onChange` will not be called. This is why Form cannot collect value from Upload.

**Comment by benjycui** [comment]
@afc163 how about triggering `onChange` before uploading?

**Comment by afc163** [comment]
@benjycui OK

**Comment by kevcal69** [comment]
@benjycui hi, im also having the same problem where im using getfielddecorator in my upload. I cant find a way to stop the upload automatically. Am i left with the option to use the manual upload without the getfielddecorator(the same with the link above for manual upload)?

**Comment by doctorrweb** [comment]
I have the same issue. I want to use antd Upload as file selector and handle the file uploading in the form with all others fields value but the req.file is undefined. Somebody can help ?.

**Comment by AmazingTurtle** [comment]
@doctorrweb I'm using `<Input type="file" />` right now. Perhaps that is sufficient for your needs?
